pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -496,12 +516,21 @@ func (gc *GarbageCollector) orphanDependents(owner objectReference, dependents [
 		go func(dependent *node) {
 			defer wg.Done()
 			// the dependent.identity.UID is used as precondition
-			patch, err := gc.deleteOwnerRefJSONMergePatch(dependent, owner.UID)
-			if err != nil {
-				errCh <- err
+			patch := deleteOwnerRefStrategicMergePatch(dependent.identity.UID, owner.UID)
+			_, err := gc.patchObject(dependent.identity, patch, types.StrategicMergePatchType)
+			if errors.IsUnsupportedMediaType(err) {
+				// StrategicMergePatch is not supported, use JSON merge patch instead
+				patch, err := gc.deleteOwnerRefJSONMergePatch(dependent, owner.UID)
+				if err != nil {
+					errCh <- fmt.Errorf("orphaning %s failed, %v", dependent.identity, err)
+					return
+				}
+				_, err = gc.patchObject(dependent.identity, patch, types.MergePatchType)
+				if err != nil {
```

Comment:
Why not "if err != nil && !errors.IsNotFound(err)" like at line 534?

Reply:
JSON merge patch cannot manipulate arrays (owner reference in this case). JSON merge patch replaces the existing array with an entire array (which we constructed in `deleteOwnerRefJSONMergePatch`). So we don't need to handle error if the target owner reference doesn't exist here.